libstuff/libstuff.cpp

```diff
@@ -2221,16 +2140,18 @@ static int _SQueryCallback(void* data, int argc, char** argv, char** colNames) {
 
 // --------------------------------------------------------------------------
 // Executes a SQLite query
-bool SQuery(sqlite3* db, const char* e, const string& sql, SQResult& result, int64_t warnThreshold) {
+int SQuery(sqlite3* db, const char* e, const string& sql, SQResult& result, int64_t warnThreshold) {
 #define MAX_TRIES 3
     // Execute the query and get the results
     uint64_t startTime = STimeNow();
     int error = 0;
+    int extErr = 0;
     for (int tries = 0; tries < MAX_TRIES; tries++) {
         result.clear();
         SDEBUG(sql);
         error = sqlite3_exec(db, sql.c_str(), _SQueryCallback, &result, 0);
-        if (error != SQLITE_BUSY) {
+        extErr = sqlite3_extended_errcode(db);
+        if (error != SQLITE_BUSY || extErr == SQLITE_BUSY_SNAPSHOT) {
```

Comment:
To confirm I understand this logic, this is saying:  "Retry the query up to three times if the database is busy (unless it's busy because the query conflicts with a different commit)"  Is this right?  If so, can you add a comment to this effect?